deploy/helm/pulumi-operator/templates/deployment.yaml

```diff
@@ -0,0 +1,108 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  {{- with .Values.deploymentAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  name: {{ include "pulumi-kubernetes-operator.fullname" . }}
+  labels:
+  {{- include "pulumi-kubernetes-operator.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  strategy:
+    type: {{ .Values.deploymentStrategy }}
+  selector:
+    matchLabels:
+    {{- include "pulumi-kubernetes-operator.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+      {{- include "pulumi-kubernetes-operator.selectorLabels" . | nindent 8 }}
+      {{- if .Values.podLabels }}
+        {{ toYaml .Values.podLabels | indent 8 }}
+      {{- end }}
+    spec:
+      containers:
+      {{- if .Values.extraSidecars }}
+      {{- toYaml .Values.extraSidecars  | nindent  8 }}
+      {{- end}}
+      - args:
+        - --zap-level=error
+        - --zap-time-encoding=iso8601
+        env:
+        {{- if .Values.extraEnv }}
+        {{- toYaml .Values.extraEnv  | nindent 10 }}
+        {{- end }}
+        - name: WATCH_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: OPERATOR_NAME
+          value: {{ include "pulumi-kubernetes-operator.name" . }}
+        - name: GRACEFUL_SHUTDOWN_TIMEOUT_DURATION
+          value: {{ .Values.controller.gracefulShutdownTimeoutDuration }}
+        - name: MAX_CONCURRENT_RECONCILES
+          value: {{ .Values.controller.maxConcurrentReconciles | quote }}
+        - name: PULUMI_INFER_NAMESPACE
+          value: {{ .Values.controller.pulumiInferNamespace | quote }}
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ .Values.controller.kubernetesClusterDomain }}
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:v{{ .Values.image.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
```

Comment:
Are we passing the args to the container? https://github.com/pulumi/pulumi-kubernetes-operator/blob/master/deploy/yaml/operator.yaml#L25-L26 We can probably set that in the values too?

Reply:
Done